amf-aml/shared/src/main/scala/amf/aml/client/scala/AMLConfiguration.scala

```diff
@@ -143,23 +143,22 @@ class AMLConfiguration private[amf] (override private[amf] val resolvers: AMFRes
 
   def withDialect(dialect: Dialect): AMLConfiguration = DialectRegister(dialect).register(this)
 
-  // TODO: what about nested $dialect references?
   def forInstance(url: String, mediaType: Option[String] = None): Future[AMLConfiguration] = {
-    val env       = predefined()
+    val newConfig = this.copy()
```

Comment:
Why make this copy? Since the config is immutable there is no worry of modifying the existing instance

Reply:
True. I will remove it.